lib/parse.js

```diff
@@ -83,6 +83,10 @@ internals.parseKeys = function (key, val, options) {
         return;
     }
 
+    // Transform dot notation to bracket notation
+
+    key = key.replace(/\.(\w+)/g, '[$1]');
```

Comment:
I have updated the regex.

What is the best way to measure perf? A few console logs show the speed to be the same as before.